### ISSUE TITLE
Id properties that are automatically generated are changed on each render

### ISSUE THREAD

**Issue by phated** [samrocksc/react-material-design-lite#19]
To test something, we put the `ReactDOM.render` call inside a `setInterval`.  When we did this, we noticed the `id`s of many of the elements were changing on each render.  Instead of generating the `id` inside the `render` method of each component, it can be generated inside the constructor if it doesn't exist on `props` and assigned to `this._id` on the instance and then used in the `render` method to avoid unnecessary DOM updates.

**Comment by samrocksc** [comment]
```
var counter = 0;

ReactDOM.render(<Tree id={counter} />, container);
setInterval(function(){
  console.log('ReRendering');
  counter += 1;
  ReactDOM.render(<Tree id={counter} />, container);
}, 2500);
```

**Comment by phated** [comment]
Closed by #20
